File: app/src/main/java/com/example/mvvm_sample/activity/MainActivity.java
package com.example.mvvm_sample.activity;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import com.example.mvvm_sample.R;
import com.example.mvvm_sample.databinding.ActivityMainBinding;
import com.example.mvvm_sample.model.SampleModel;
import com.example.mvvm_sample.viewmodel.SampleAndroidViewModelFactory;
import com.example.mvvm_sample.viewmodel.SampleViewModel;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setLifecycleOwner(this);

        SampleModel sampleModel = new SampleModel();
        SampleAndroidViewModelFactory sampleAndroidViewModelFactory = new SampleAndroidViewModelFactory(getApplication(), sampleModel);
        SampleViewModel viewModel = new ViewModelProvider(this, sampleAndroidViewModelFactory).get(SampleViewModel.class);

        binding.setSampleViewModel(viewModel);

    }
}

File: app/src/main/java/com/example/mvvm_sample/activity/MainActivity.kt
package com.example.mvvm_sample.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.example.mvvm_sample.MyApplication
import com.example.mvvm_sample.R
import com.example.mvvm_sample.adapter.ArticleAdapter
import com.example.mvvm_sample.databinding.ActivityMainBinding
import com.example.mvvm_sample.model.SampleModel
import com.example.mvvm_sample.viewmodel.SampleAndroidViewModelFactory
import com.example.mvvm_sample.viewmodel.SampleViewModel
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {
    private val binding: ActivityMainBinding by lazy {
        DataBindingUtil.setContentView(
            this, R.layout.activity_main
        )
    }
    private val viewModel by lazy {
        ViewModelProvider(
            this, SampleAndroidViewModelFactory(MyApplication(), SampleModel())
        )[SampleViewModel::class.java]
    }
    private val adapter by lazy {
        ArticleAdapter(this, viewModel)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.sampleViewModel = viewModel
        binding.lifecycleOwner = this
        binding.lvArticleList.adapter = adapter
        viewModel.postResponse.observe(this) {
            adapter.notifyDataSetChanged()
        }
    }

}